microsoft-edge/devtools-guide-chromium/evaluate-performance/unminify.md

```diff
@@ -0,0 +1,96 @@
+
+1. In the **Performance** tool, click in the main area (the flame chart), and use the mouse wheel to zoom in.  Click and drag left and right.  Look at the function names in the flame chart.  Select a small section of the profile and look at the flame chart.  Note the timestamp range.
+
+   The minified flame chart in the **Performance** tool shows terse function names, such as **XI**, **Gi**, **d**, **o**, **an**, **c**, **co**, and **a**:
+
+   ![The minified flame chart in the performance profile.](images/minified-perf-profile.msft.png)
+
+
+<!-- ====================================================================== -->
+## Step 3 - Unminify the code
+
+1. Click the **Unminify** (![The Unminify button icon.](images/unminify-icon.png)) button:
+
+   ![The Unminify button in the Performance tool.](images/perf-profile-unminify-button.msft.png)
+
+   In DevTools, a popup briefly appears, reading: **Status: creating unminfied file**.  Then the **Save As** dialog box opens.  The default filename is a pattern starting with date and time like `yymmddThhmmss`, such as: `Profile-20220307T164948-unminified.json`.  The default directory is the **Downloads** directory.
+
+1. Select a filename and directory and then click the **Save** button.
+
+   The **Performance** tool creates a new version of the recorded performance profile, with some function names from the flame chart replaced by more meaningful function names by using your source maps.  Some of the minified production names might still appear, because source maps don't always provide the necessary information for the **Performance** tool to map all function names.
+
+
+<!-- ====================================================================== -->
+## Step 4 - Load the unminified profile
+
+1. In the **Performance** tool, click the **Load profile** (![The 'Load profile` icon button in the Performance tool.](images/load-profile-icon.png)) icon button.  Or, right-click a blank area of the **Performance** tool, and then select **Load profile**.
+
+   ![Load Profile.](../media/evaluate-performance-performance-refreshed-disable-javascript-samples-checkbox-off-load-profile.msft.png)
+
+1. In the **Open** dialog box, select the unminified performance profile `.json` file that you just downloaded.  If needed, for details, see [Load a recording](/microsoft-edge/devtools-guide-chromium/evaluate-performance/reference#load-a-recording) in _Performance features reference_, and then continue below.
+
+1. Select the same section (timestamp range) of the profile you looked at in the first image.
+
+   Some function names are now unminified.  The unminified flame chart in the **Performance** tool now shows those functions labeled with more meaningful names taken from your source maps, such as **gg (XI)**, **oh (Gi)**, **useObserver (d)**, **useQueuedForceUpdateBlock (o)**, **trackDerivedFunction (an)**, **initializeIconColorMap (c)**, **getExtraToolControls (co)**, and **isToggleEnabled (a)**.
```

Comment:
It's not really that helpful that some of these function names are still minified, just with different names. It's important to remember that there are multiple transforms between source code and production code and sourcemaps may undo one of those transformations but not all of them perfectly. I'd cut the ones that don't clearly map to unminified function names.
```suggestion
   Some function names are now unminified.  The unminified flame chart in the **Performance** tool now shows those functions labeled with more meaningful names taken from your source maps, such as **useObserver (d)**, **useQueuedForceUpdateBlock (o)**, **trackDerivedFunction (an)**, **initializeIconColorMap (c)**, **getExtraToolControls (co)**, and **isToggleEnabled (a)**.
```